templates/buildkite-elastic.yml

```diff
@@ -378,15 +378,21 @@ Resources:
     Type: AWS::IAM::Role
     Properties:
       RoleName: { "Fn::If": [ SetInstanceRoleName, { Ref: InstanceRoleName }, { "Fn::Sub": "${AWS::StackName}-Role" } ] }
-      ManagedPolicyArns:
-        - "Fn::If":
-          - UseManagedPolicyARN
-          - { Ref: ManagedPolicyARN }
-          - { Ref: 'AWS::NoValue' }
-        - "Fn::If":
-          - UseECR
-          - { "Fn::FindInMap": [ ECRManagedPolicy, { Ref: ECRAccessPolicy }, 'Policy' ] }
-          - { Ref: 'AWS::NoValue' }
+      # Support multiple policies to attach by merging the values together and splitting on ','
+      ManagedPolicyArns: !Split
```

Comment:
Per the PR discussion, this produces an empty string (thus not an ARN) when no policies are to be attached -- thus needs more thought :(